### DISCUSSION TITLE
Texture Objects when doing Off-Screen Rendering

### DISCUSSION THREAD

**Discussion by Mattc2021** [pelicanmapping/osgearth#2244] [Support Q&A] [answered]
Given an osg earth viewer, how would I render that off screen? Attaching an camera with a custom color attachment doesn't seem to cause a texture object to generate for the texture, am I supposed to manually create the texture object when trying to do off-screen rendering?

**Comment by gwaldron** [comment, marked as answer]
Hi Matt. Please look at the [osgearth_mrt.cpp example](https://github.com/gwaldron/osgearth/blob/master/src/applications/osgearth_mrt/osgearth_mrt.cpp). It renders the scene to a texture (actually a series of textures) and then renders those textures to the framebuffer in another pass. The `createMRTPass` function renders the texture(s), and the `createFramebufferPass` renders those textures to a quad.

**Comment by Mattc2021** [comment]
Would this also work instead of using the app we rendered that texture embedded into a OpenGL3/GLFW Imgui window? I see the viewer.run(), but I'm assuming we can use viewer.frame() as well to just get the quad and render them on a target application window.
